### ISSUE TITLE
Bug after updating to 1.7: Can't save relationship entity

### ISSUE THREAD

**Issue by dugajean** [zenstruck/foundry#114]
I updated to 1.7 with Symfony 5.2.2. Before this update, all my tests were passing, however after this update, Foundry is unable to create an entity with a relationship.

My code looks like this:

```
$department1 = DepartmentFactory::new([
    'company' => $company1,
    'name' => 'Human Resources',
])->create();

$department2 = DepartmentFactory::new([
    'company' => $company2,
    'name' => 'Human Resources',
])->create();
```

Both `$company1` and `$company2` are saved in the database (both are also created via a `CompanyFactory`), however, I get this error:

> 1) App\Tests\Functional\DepartmentResourceTest::test_fetching_departments_items_with_same_slug_authenticated
Doctrine\ORM\ORMInvalidArgumentException: A new entity was found through the relationship 'App\Entity\Department#company' that was not configured to cascade persist operations for entity: App\Entity\Company@000000001f917ea1000000007f4b7464. To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist this association in the mapping for example @ManyToOne(..,cascade={"persist"}). If you cannot find out which entity causes the problem implement 'App\Entity\Company#__toString()' to get a clue.

I tried adding `cascade={"persist"}` to my entity, but this isn't necessary for my business logic and it yielded other errors. The surprising thing is that this wasn't happening before the update.

Any ideas?

**Comment by kbond** [comment]
Hey @dugajean what was the version you upgraded from?

**Comment by dugajean** [comment]
@kbond I upgraded from 1.4.0

**Comment by kbond** [comment]
Ok, I've seen this error before - I believe it's because the em is cleared between creating the company's and creating the departments. Since `$company1` & `$company2` are no longer managed by doctrine, they appear to Doctrine as "new" objects.

Can you share some more context from your test? what happens between creating the company's and departments?

**Comment by kbond** [comment]
@dugajean I was able to duplicate the error in #114 and add a possible fix. Are you able to confirm this works for you?

**Comment by dugajean** [comment]
Hey @kbond, thanks for the quick fix! 

I was able to get around the problem on my own by first doing all factory creations and then making API requests after. But I'll try your update and see if that helps.

My test looked as follows:

1. Generate some entities
2. Make an API request to fetch an access token
3 Generate some more entities after
4. Make more API requests with the access token I fetched earlier
5. Make assertions

I swapped 2 and 3 to make things work. It resulted in some ugly code which I need to refactor, but it works.

**Comment by kbond** [comment]
Ah yes, 2 must be clearing the em.

Assuming `$company1/2` are proxies, I believe [enabling auto refreshing](https://github.com/zenstruck/foundry#auto-refresh) would also solve the issue.

**Comment by kbond** [comment]
@dugajean, have you had a chance to confirm #117 works for your use case?

**Comment by dugajean** [comment]
Didn't have the chance to test this yet.

**Comment by kbond** [comment]
Ok, no worries. I merged that fix and released 1.7.1. Hopefully it fixes your problem and we can reopen this issue if not.
